crates/forge/tests/cli/test_cmd.rs

```diff
@@ -2097,5 +2097,34 @@ contract JunitReportTest is Test {
 </testsuites>
 
 
+"#]]);
+});
+
+forgetest_init!(test_deprecated_cheatcode_err, |prj, cmd| {
+    prj.add_test(
+        "DeprecatedCheatcodeTest.t.sol",
+        r#"
+        import "forge-std/Test.sol";
+        contract DeprecatedCheatcodeTest is Test {
+            function test_deprecated_cheatcode() public view {
+                vm.keyExists('{"a": 123}', ".a");
+            }
+        }
+   "#,
+    )
+    .unwrap();
+
+    cmd.args(["test", "--mt", "test_deprecated_cheatcode"]).assert_success().stdout_eq(str![[r#"
+[COMPILING_FILES] with [SOLC_VERSION]
+[SOLC_VERSION] [ELAPSED]
+Compiler run successful!
+[..] ERROR foundry_cheatcodes: keyExists(string,string) cheatcode is being deprecated and will be removed in future versions.
```

Comment:
Ideally we would accompany this log by the replacing equivalent if applicable

Reply:
we could be adding the cheatcode description but this assume a certain format. I think ideally the `Status::Deprecated` would accept the replacement cheatcode but this implies some more changes to `Status` enum and not sure it worth the effort, wdyt?

Reply:
I think it is important to directly point users to the replacing cheatcode as it makes fixing it much easier for users when they encounter the warning

> ideally the Status::Deprecated would accept the replacement cheatcode but this implies some more changes to Status enum

I think this is the correct approach and worth doing

Reply:
kk, makes sense, will look into adding such

Reply:
added such in https://github.com/foundry-rs/foundry/pull/8883/commits/597a2e3bc7f9a69a9f732090cbf8350c68df6951 with a new attr `replacement` (`#[cheatcode(group = Json, status = Deprecated, replacement = "keyExistsJson")]`) since replacement is optionally 

![image](https://github.com/user-attachments/assets/959a6014-806d-4f11-916b-505f7c7cb01e)

hope this makes sense

Reply:
Looks good, implementation makes sense to me